xlators/storage/posix/src/posix-inode-fd-ops.c

```diff
@@ -1539,6 +1540,18 @@ posix_truncate(call_frame_t *frame, xlator_t *this, loc_t *loc, off_t offset,
         goto out;
     }
 
+    if (prebuf.ia_size < offset) {
```

Comment:
if you want this to be completely fool proof, then you should look at number of blocks in prebuf and postbuf. 

Usecase: `truncate -s 100G`; (succeeds without issues), next do dd with `oflag=notrunc`(on a 50GB brick).

Reply:
Ack